perfact/zodbsync/zodbsync.py

```diff
@@ -784,7 +786,11 @@ def _playback_path(self, pathinfo):
             srv_contents = obj_contents(obj) if obj else []
 
             # Find IDs in Data.fs object not present in file system
-            del_ids = [a for a in srv_contents if a not in contents]
+            del_ids = [
+                a for a in srv_contents
+                if a not in contents and
+                (path != '/acl_users' and a != 'acl_users')
```

Comment:
Why doesn't this prevent the deletion of `/some_module/acl_users` in recursive mode? I'm a bit confused...

Reply:
I think it does prevent it, but the test does not notice because of some transaction shenanigans. I stepped with the debugger into the `playback` and noticed that both `fs_contents` and `srv_contents` were empty, so nothing was deleted. Not sure why the first check for the presence of `acl_users` in the tests does not fail though...

Reply:
To explain: If we simply manipulate something with directly accessing `self.app`, this automatically opens a transaction. When the next `self.run(...)` is executed, this usually also starts a transaction, aborting the already open transaction. Therefore the creation of `/some_module` and `/some_module/acl_users` was never commited, but visible in the first check (since it was still the same transaction), but no longer visible once the `playback` started.

Reply:
I guess `playback` explicitly starts a new transaction (to be able to roll it back later) while `record` does not (because it does not expect to change anything inside the Data.FS).